File: member/src/main/java/com/luckytree/member/member/adapter/jpa/MemberRepository.java
package com.luckytree.member.member.adapter.jpa;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<MemberEntity, Long> {

    Optional<MemberEntity> findByIdAndEmail(Long id, String email);
}

File: auth/src/main/kotlin/com/example/auth/persistence/MemberRepository.kt
package com.example.auth.persistence

import com.example.auth.model.entity.Member
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {
    @EntityGraph(attributePaths = ["authorities"])
    fun findOneWithAuthoritiesById(id: Long): Optional<Member>
}
